### PR TITLE
Fix: added oppia android link to oppiabot comment

### PR DESCRIPTION
**Pull request by nlok5923**
Fixes: #256 

## Explanation

added oppia android set up link to oppiabot comment 

## Checklist
- [ ] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

**Comment by nlok5923** [comment]
@vojtechjelinek @jameesjohn please review this pull request.

File: lib/apiForSheets.js [hunk 1]
```diff
@@ -103,10 +107,11 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
...
         linkResult +
-        " to get started? You'll need to do " +
-        'this before we can accept your PR. I am closing this PR for now. ' +
-        'Feel free to re-open it once you are done with the above ' +
-        'instructions. Thanks!',
+        " to get started with oppia and "  +  linkOppiaAndroid +
+        " to get started with oppia-android? You'll need to do this " +
+        "before we can accept your PR. I am closing this PR for now. " +
+        "Feel free to re-open it once you are done with the above " +
+        "instructions. Thanks!",
```
Comment (jameesjohn) [inline lib/apiForSheets.js:114]: Can you use an if statement to check if the PR is from the oppia android repo and then respond with different comments for the repos?

File: lib/apiForSheets.js [hunk 2]
```diff
@@ -95,6 +95,10 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
       'https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#' +
```
Comment (jameesjohn) [inline lib/apiForSheets.js:95]: The tests need to be updated following this change.

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks @nlok5923, I left some comments.

File: lib/apiForSheets.js [hunk 3]
```diff
@@ -95,6 +95,10 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
...
       'setting-things-up'
     );
+    var oppiaAndroidLinkText = 'here';
+    var linkOppiaAndroid = oppiaAndroidLinkText.link(
+    'https://github.com/oppia/oppia-android/wiki#onboarding-instructions'
+    );
```
Comment (vojtechjelinek) [inline lib/apiForSheets.js:101]:
```suggestion
    var linkOppiaAndroid = oppiaAndroidLinkText.link(
      'https://github.com/oppia/oppia-android/wiki#onboarding-instructions'
    );
```

File: lib/apiForSheets.js (hunk 1 above)
Comment (nlok5923) [reply lib/apiForSheets.js]: addressed in latest commit.

File: lib/apiForSheets.js (hunk 2 above)
Comment (nlok5923) [reply lib/apiForSheets.js:96]: working on tests

File: lib/apiForSheets.js (hunk 3 above)
Comment (nlok5923) [reply lib/apiForSheets.js]: changes addressed.

File: lib/apiForSheets.js (hunk 2 above)
Comment (nlok5923) [reply lib/apiForSheets.js:96]: tests updated.

**Comment by vojtechjelinek** [comment]
@nlok5923 Please make sure to assign the reviewers to the PR when you want a review (you should be able to do that now that you're a contributor).

**Comment by nlok5923** [comment]
@vojtechjelinek still i am not able to assign reviewers 
image

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Apologies for the late review.
You need also to add a test case where the source repo of the pull request is oppia-android.

File: lib/apiForSheets.js [hunk 4]
```diff
@@ -43,7 +43,8 @@ module.exports.authorize = function (credentials) {
   return oauth2Client;
 };
 
-module.exports.generateOutput = async function (rows, prNumber, prDetails) {
+module.exports.generateOutput = async function (rows, prNumber, prDetails,
+    repositoryName) {
```
Comment (vojtechjelinek) [inline lib/apiForSheets.js:47]:
```suggestion
module.exports.generateOutput = async function (
    rows, prNumber, prDetails, repositoryName) {
```

File: lib/apiForSheets.js [hunk 5]
```diff
@@ -95,19 +96,38 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
...
+    if (repositoryName === 'oppia-android') {
+      commentBody = 'Hi! ' +
+      userNamesNotSignedCLA +
+      'Welcome to Oppia! Please could you ' +
+      'follow the instructions ' +
+      linkOppiaAndroid +
+      ' to get started with oppia-android? You\'ll need to do this ' +
+      'before we can accept your PR. I am closing this PR for now. ' +
+      'Feel free to re-open it once you are done with the above ' +
+      'instructions. Thanks!';
```
Comment (vojtechjelinek) [inline lib/apiForSheets.js:114]:
```suggestion
      commentBody = (
        'Hi! ' + userNamesNotSignedCLA +
        'Welcome to Oppia! Please could you follow the instructions ' + 
        linkOppiaAndroid + ' to get started with oppia-android? ' +
        'You\'ll need to do this before we can accept your PR. I am ' +
        closing this PR for now. Feel free to re-open it once you ' +
        are done with the above instructions. Thanks!'
      );
```

Comment (vojtechjelinek) [reply lib/apiForSheets.js:114]: ditto below

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
Thanks! Left a few comments.

**Comment by nlok5923** [comment]
> Apologies for the late review.
> You need also to add a test case where the source repo of the pull request is oppia-android.

yes @jameesjohn i had checked first the repo over which pr is created using the if statement and then commenting corresponding message.

File: lib/apiForSheets.js [hunk 6]
```diff
@@ -95,19 +96,38 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
       'https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#' +
       'setting-things-up'
     );
+    var oppiaAndroidLinkText = 'here';
+    var linkOppiaAndroid = oppiaAndroidLinkText.link(
+      'https://github.com/oppia/oppia-android/wiki#onboarding-instructions'
+    );
+
+    var commentBody = '';
+    if (repositoryName === 'oppia-android') {
```
Comment (nlok5923) [inline lib/apiForSheets.js:105]: here i had checked whether pr is made on oppia android repo.

**Review by nlok5923 — COMMENTED** [review]
please refer to line 105

File: lib/apiForSheets.js (hunk 4 above)
Comment (nlok5923) [reply lib/apiForSheets.js:47]: yes addressed in latest commit.

File: lib/apiForSheets.js (hunk 5 above)
Comment (nlok5923) [reply lib/apiForSheets.js:114]: yes addressed in latest commit.

File: lib/apiForSheets.js (hunk 6 above)
Comment (nlok5923) [reply lib/apiForSheets.js:105]: will add test case of source repo as oppia-android in latest commit.

**Comment by nlok5923** [comment]
@jameesjohn i think the current solution also cover the test case if pr is from source repo please let me know if i missed something. as `repository.name` property contains repo name.

File: lib/apiForSheets.js [hunk 7]
```diff
@@ -43,7 +43,8 @@ module.exports.authorize = function (credentials) {
   return oauth2Client;
 };
 
-module.exports.generateOutput = async function (rows, prNumber, prDetails) {
+module.exports.generateOutput = async function (
+  rows, prNumber, prDetails, repositoryName) {
```
Comment (vojtechjelinek) [inline lib/apiForSheets.js:47]:
```suggestion
    rows, prNumber, prDetails, repositoryName) {
```

File: lib/apiForSheets.js [hunk 8]
```diff
@@ -95,19 +96,36 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
       'https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#' +
       'setting-things-up'
     );
+    var oppiaAndroidLinkText = 'here';
+    var linkOppiaAndroid = oppiaAndroidLinkText.link(
+      'https://github.com/oppia/oppia-android/wiki#onboarding-instructions'
+    );
```
Comment (vojtechjelinek) [inline lib/apiForSheets.js:102]:
```suggestion
    var linkOppiaAndroid = 'onboarding instructions'.link(
      'https://github.com/oppia/oppia-android/wiki#onboarding-instructions'
    );
```

File: lib/apiForSheets.js [hunk 9]
```diff
@@ -95,19 +96,36 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
...
+
+    var commentBody = '';
+    if (repositoryName === 'oppia-android') {
+      commentBody = (
+        'Hi! ' + userNamesNotSignedCLA +
+        'Welcome to Oppia! Please could you follow the instructions ' + 
+        linkOppiaAndroid + ' to get started with oppia-android? ' +
```
Comment (vojtechjelinek) [inline lib/apiForSheets.js:109]:
```suggestion
        'Welcome to Oppia! Please could you follow the ' + 
        linkOppiaAndroid + ' to get started with oppia-android? ' +
```

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
THanks! Left a few comments.

File: lib/apiForSheets.js (hunk 9 above)
Comment (nlok5923) [reply lib/apiForSheets.js]: addressed in latest commit.

File: lib/apiForSheets.js (hunk 8 above)
Comment (nlok5923) [reply lib/apiForSheets.js]: addressed.

File: lib/apiForSheets.js (hunk 7 above)
Comment (nlok5923) [reply lib/apiForSheets.js]: addressed in latest commit.

File: spec/apiForSheetsSpec.js [hunk 10]
```diff
@@ -264,17 +264,37 @@ describe('Api For Sheets Module', () => {
...
-          number: pullRequestPayload.payload.pull_request.number,
-          body:
+        var oppiaAndroidLinkText = 'here';
+        var linkOppiaAndroid = oppiaAndroidLinkText.link(
+          'https://github.com/oppia/oppia-android/wiki#onboarding-instructions'
+        );
+
+        var commentBody = '';
+        if (pullRequestPayload.payload.repository.name === 'oppia-android') {
+          commentBody = 'Hi! ' +
```
Comment (jameesjohn) [inline spec/apiForSheetsSpec.js:274]: This is not the way to do this.
You need to update the repository in the request payload to be oppia-android in a different describe block. Then assert that the correct comment is being sent.

Comment (nlok5923) [reply spec/apiForSheetsSpec.js]: @jameesjohn please can you review the test as i had added different describe block to set repository as oppia android.

File: spec/apiForSheetsSpec.js [hunk 11]
```diff
@@ -250,39 +250,81 @@ describe('Api For Sheets Module', () => {
...
+        repo: pullRequestPayload.payload.repository.name,
+        state: 'closed',
+      });
+    });
+
+    describe('when pull request created at oppia-android repository', () => {
+      beforeEach(function (done) {
+        pullRequestPayload.payload.repository.name = 'oppia-android';
+        done();
+      });
```
Comment (jameesjohn) [inline spec/apiForSheetsSpec.js:292]: Add an `afterEach` block that reverts the repository back to its original state.

File: spec/apiForSheetsSpec.js [hunk 12]
```diff
@@ -250,39 +250,81 @@ describe('Api For Sheets Module', () => {
...
+
+      it('should close the PR if atleast one user has not signed cla',
+        async () => {
+        // Add username to CLA sheet.
+          await apiForSheetsModule.generateOutput(
+            [...claData, ['testuser7777']],
+            pullRequestPayload.payload.pull_request.number,
+            pullRequestPayload.payload.pull_request
+          );
```
Comment (jameesjohn) [inline spec/apiForSheetsSpec.js:301]: This function expects the repository name to be passed as well.

File: lib/apiForSheets.js [hunk 13]
```diff
@@ -95,19 +96,31 @@ module.exports.generateOutput = async function (rows, prNumber, prDetails) {
...
+    );
+
+    var commentBody = '';
+    if (repositoryName === 'oppia-android') {
+      commentBody = 'Hi! ' + userNamesNotSignedCLA +
+        'Welcome to Oppia! Please could you follow the ' +
+        linkOppiaAndroid + ' to get started with oppia-android? ' +
+        'You\'ll need to do this before we can accept your PR. I am ' +
+        'closing this PR for now. Feel free to re-open it once you ' +
+        'are done with the above instructions. Thanks!'
```
Comment (jameesjohn) [inline lib/apiForSheets.js:110]: The comment generated here is different from that being asserted in your test. PTAL.

**Review by jameesjohn — CHANGES_REQUESTED** [review]

File: lib/apiForSheets.js (hunk 13 above)
Comment (nlok5923) [reply lib/apiForSheets.js:110]: working on requested changes.

**Comment by nlok5923** [comment]
@jameesjohn please review this pr now all test are passing locally.
image

File: spec/apiForSheetsSpec.js (hunk 12 above)
Comment (nlok5923) [reply spec/apiForSheetsSpec.js:308]: addressed in the latest commit

File: spec/apiForSheetsSpec.js (hunk 11 above)
Comment (nlok5923) [reply spec/apiForSheetsSpec.js:293]: addressed in the latest commit

File: lib/apiForSheets.js (hunk 13 above)
Comment (nlok5923) [reply lib/apiForSheets.js]: addressed all changes in the latest commit.

**Review by jameesjohn — APPROVED** [review]
LGTM

**Comment by jameesjohn** [comment]
PTAL @vojtechjelinek

**Review by vojtechjelinek — APPROVED** [review]
LGTM!

**Pull request merged**